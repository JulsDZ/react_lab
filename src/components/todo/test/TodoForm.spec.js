import React from 'react';
import { shallow, mount, render, configure  } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import {Â TodoForm } from '../';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const output = shallow(
    <TodoForm  
        currentTodo={{id: 1, name: 'Learn JSX', isComplete: true}}
    />
  );
  //handleInputChange={this.handleInputChange}
  //currentTodo={this.state.currentTodo}
  //handleSubmit={submitHandler}
describe('TodoForm', () => {
  it('should render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should the input element be rendered', () => {
    expect(output.containsAllMatchingElements([
        <input />
      ])).toEqual(true);
  });
});
