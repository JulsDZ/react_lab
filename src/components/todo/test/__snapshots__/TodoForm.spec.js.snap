// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm should  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        onChange={undefined}
        type="text"
        value={undefined}
/>,
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "onChange": undefined,
        "type": "text",
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          onChange={undefined}
          type="text"
          value={undefined}
/>,
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": undefined,
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TodoForm should have values in it 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        onChange={undefined}
        type="text"
        value={undefined}
/>,
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "onChange": undefined,
        "type": "text",
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          onChange={undefined}
          type="text"
          value={undefined}
/>,
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": undefined,
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TodoForm should render correctly 1`] = `
<form>
  <input
    type="text"
    value={
      Object {
        "id": 1,
        "isComplete": true,
        "name": "Learn JSX",
      }
    }
  />
</form>
`;
